### ISSUE TITLE
Packet queue performance can be improved

### ISSUE THREAD

**Issue by nhuang-tt** [tenstorrent/tt-metal#15474]
The base class of the routing kernels, `packet_queue`, has a lot of branches and accesses data members at runtime that can be otherwise done at compile time. Just running `test_tx_rx` (Traffic generator to receiver bandwidth test), the baseline is 0.84 B/cycle and wall clock bandwidth is 22.86 MB/s. After changing the `if` statements to compile time it increased to 0.95 B/cycle and 26.45 MB/s.

**Comment by davorchap** [comment]
@ubcheema sync with Nigel

**Comment by nhuang-tt** [comment]
Yes, I will make the updates after the dependency: [power of 2 ring buffer size ticket](https://github.com/tenstorrent/tt-metal/issues/14427)
